sherlock/internal/clients/google_bucket/client.go

```diff
@@ -0,0 +1,75 @@
+package google_bucket
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+type GcsClientActual struct {
+	GcsClient *storage.Client
+}
+
+func InitializeStorageClient(ctx context.Context, impersonateAccount ...string) (*GcsClientActual, error) {
+	// Client uses xml google api's by default but docs reccomend using JSON - will become default behavior in later release
+	client, err := storage.NewClient(ctx, storage.WithJSONReads())
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create storage client: %v\n", err)
+	}
+	return &GcsClientActual{GcsClient: client}, err
+}
+
+func (c *GcsClientActual) ListBlobs(ctx context.Context, bucket string) ([]*storage.ObjectAttrs, error) {
```

Comment:
not sure if go is enforcing this but please don't use single letter variables. Even in small methods it can add cognitive overhead e.g. in this case, is c == client? context? something else? it's not hard to find out but it breaks up readability.